phase/detector.go

```diff
@@ -192,26 +180,29 @@ func (d *Detector) detectGroup(group buildpack.Group, done []buildpack.GroupElem
 				// FIXME: cyclical references lead to infinite recursion
 				return d.detectOrder(order, done, group.Group[i+1:], groupEl.Optional, wg)
 			}
-			descriptor = bpDescriptor // standardize the type so below we don't have to care whether it was an extension
+			descriptor = bpDescriptor // Standardize the type so below we don't have to care whether it is an extension.
 		} else {
 			descriptor, err = d.DirStore.LookupExt(groupEl.ID, groupEl.Version)
 			if err != nil {
 				return nil, nil, err
 			}
 		}
+
+		// Check target compatibility.
 		if d.PlatformAPI.AtLeast("0.12") {
-			targetMatch := false
-			if isWildcard(d.AnalyzeMD.RunImageTarget()) || hasWildcard(descriptor.TargetsList()) {
+			var targetMatch bool
+			if len(descriptor.TargetsList()) == 0 {
```

Comment:
Is this for backwards-compatibility with buildpacks that don't have a `targets` field? Is this spec-compliant?

Reply:
In reality, this is for tests. In practice, the lifecycle will infer at least an OS when targets are missing. Here is what the spec has to say about matching:

> If the targets list is empty, tools reading buildpack.toml will assume:

> os = "linux" and arch = <matches any> if ./bin/build is present
os = "windows" and arch = <matches any> if ./bin/build.bat or ./bin/build.exe are present
Metadata specified in [[targets]] is validated against the runtime and build-time base images.

> A buildpack target satisfies a base image target when os, arch, and variant match and at least one distribution in distros (if provided) matches